### DISCUSSION TITLE
GRAM output has different Pixel width

### DISCUSSION THREAD

**Discussion by me-nkr** [hneemann/Digital#1277] [Q&A]
![image](https://github.com/hneemann/Digital/assets/60778845/587fe249-1043-492c-bf01-b18737f741c9)

This image should explain the problem, and the circuit is attached below

[GRAM_Mapper.zip](https://github.com/hneemann/Digital/files/14969691/GRAM_Mapper.zip)

- The circuit is something that takes one word and writes it to 16 addresses in the GRAM in one clock cycle.
-  The input word is set to a constant `0b0001000000001000`
- Since the GRAM is 512x256px in resolution, there's a dot at each 4th and 13th pixel, which in turn draws the vertical line
- **It seems like the dot at the 4th pixel is thinner than the dot at the 13th pixel.**

The circuit is working fine in terms of functionality, but just wanted to see if it's a problem with my circuit or a potential bug.

**Comment by mengstr** [comment]
For me it works just fine.
<img width="196" alt="Screenshot 2024-04-14 at 11 28 48" src="https://github.com/hneemann/Digital/assets/325326/2c55ff3b-be0a-4556-ade0-c61be32d943d">

Did you change the size of the "video monitor" window? When I try your design it works perfectly giving nice even lines. But if I drag/adjust the window to a size that is not an exact multiple of the designs output video resolution it gets wonky.  

And that is to be expected. Like if you want to draw 512 pixels on an area that is 600 pixels wide, then some pixels must be double width.  And the same thing if you shrink the window size to, say 500 pixels, then 12 of the lines can't be drawn so there will be strange gaps in the output.

If I resize the window, stop the running, and then restart it the last window size if remembered. But if you close Digital and then load your design again the video-window will have its correct size again.

<img width="584" alt="Screenshot 2024-04-14 at 11 30 00" src="https://github.com/hneemann/Digital/assets/325326/ca3c2a3b-ecf5-4bf6-be82-582bff466e27">
<img width="375" alt="Screenshot 2024-04-14 at 11 29 49" src="https://github.com/hneemann/Digital/assets/325326/b6389dbb-2ec3-4dd1-a92b-8a239f768077">
<img width="368" alt="Screenshot 2024-04-14 at 11 29 39" src="https://github.com/hneemann/Digital/assets/325326/df37fc68-e7f6-4ba3-be32-3581e4e9e52a">
<img width="306" alt="Screenshot 2024-04-14 at 11 29 27" src="https://github.com/hneemann/Digital/assets/325326/b04cb0ca-3e14-49ca-bd3e-13a83353c129">

**Reply by me-nkr** [reply]
I did not resize the window and to make sure I did restart digital again and again. Also, I am noticing it is happening only for the pixels that is in the right half of each 16-pixel.

**Reply by hneemann** [reply]
What happens if you resize the window?

**Comment by xotmatrix** [comment]
It sure looks like Windows has application scaling set to 150%, although when I do it the results are anti-aliased.
<img width="385" alt="javaw_2024-04-16_01-56-11" src="https://github.com/hneemann/Digital/assets/14209329/160598b0-3f10-42a6-af12-1301007ce58a">

**Reply by me-nkr** [reply]
I do have my display set to 150%, is it the reason for the problem?
